src/templates/angular/src/app/pages/header/header.component.ts

```diff
@@ -1,15 +1,32 @@
 import { Component, OnInit } from '@angular/core';
-
+<% if(isDark){%> import { FormBuilder, FormGroup } from '@angular/forms';
+const $body = document.body; <%}%>
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.css']
 })
 export class HeaderComponent implements OnInit {
-
-  constructor() { }
-
+  <% if(isDark){%> isChecked: boolean = true;
+  headerForm!: FormGroup;<%}%>
+  constructor( <% if(isDark){%> private fb: FormBuilder <%}%>) { }
   ngOnInit(): void {
+    <% if(isDark){%>
+    let isThemeSelected: any = sessionStorage.getItem('isDarkModeSelected');
+    let selected = JSON.parse(isThemeSelected);
+    if(selected) $body.setAttribute('data-theme', 'dark')
+    else $body.setAttribute('data-theme', 'light');
+    this.initForm(selected)  <%}%>
```

Comment:
let's avoid, using a whole chunk of code in conditional statements,
create functions/methods & use them to invoke based on conditional variables.

Reply:
we can keep this whole chunk of code outside of ngOnInit